docs/docs/comet-core-development/creating-customizable-admin-components.md

```diff
@@ -0,0 +1,239 @@
+
+## Basics
+
+Generally, a Comet Admin component should export the following in the `index.ts` file:
+
+-   The component itself, with a unique name
+-   The props type, named as the component, followed by `Props`
+-   The class key type, named as the component, followed by `ClassKey`
+
+```tsx
+export { MyComponent, MyComponentProps, MyComponentClassKey } from "./src/MyComponent";
+```
+
+## Class keys
+
+A components `ClassKey` type defines all the individual elements (or slots) of a component or a modifier/alternative style of a component.
+
+The outermost slot of a component is generally called `root`.
+
+```tsx
+export type MyComponentClassKey = "root" | "title";
+```
+
+## Slots
+
+Each element or subcomponent of a Comet Admin component is defined as a slot.
+
+A slot is created and styled by using `createComponentSlot()` and passing in the HTML element or component you want to be the base of your slot.
+
+Additionally, the component's name must be passed in, as well as the slot's name, as defined in the class key type.
+
+```tsx
+const Root = createComponentSlot("div")<MyComponentClassKey>({
+    componentName: "MyComponent",
+    slotName: "root",
+})(
+    ({ theme }) => css`
+        background-color: ${theme.palette.primary.main};
+    `,
+);
+
+const Title = createComponentSlot(Typography)<MyComponentClassKey>({
+    componentName: "MyComponent",
+    slotName: "title",
+})(css`
+    color: lime;
+`);
+```
+
+## Defining the props type
+
+The props type of Comet Admin components must extend `ThemedComponentBaseProps`.
+As a generic, an object must be passed in with the slot's name as the keys and the type of their base element or component as values.
+
+```tsx
+export interface MyComponentProps
+    extends ThemedComponentBaseProps<{
+        root: "div";
+        title: typeof Typography;
+    }> {
+    variant?: "primary" | "secondary";
+    children?: React.ReactNode;
+}
+```
+
+## Using props
+
+Props should not be used directly but through the `useThemeProps` hook instead.
+This ensures that the components `defaultProps` from the theme are merged with the passed-in props.
+
+Note that the name passed to the `useThemeProps` hook must be the same as the component's name, prefixed with `CometAdmin`.
+
+```tsx
+export function MyComponent(inProps: MyComponentProps) {
+    const { children, variant } = useThemeProps({ props: inProps, name: "CometAdminMyComponent" });
+    return (
+        // ...
+    )
+}
+```
+
+## `slotProps` and `restProps`
+
+To allow overriding the props of each slot, the `slotProps` object should be spread into each slot.
+Additionally, the `sx` and `className` props should be passed into the root slot, for example with `...restProps`.
+
+```tsx
+export function MyComponent(inProps: MyComponentProps) {
+    const { slotProps, ...restProps } = useThemeProps({
+        props: inProps,
+        name: "CometAdminMyComponent",
+    });
+    return (
+        <Root {...slotProps?.root} {...restProps}>
```

Comment:
Should the slotProps always be the first prop?

Reply:
Generally `slotProps` and `restProps` should be last, so props defined inside the component can be overridden. 
Exceptions would be props that should never be overridden. 

I will add this info to the docs as well 👍🏼